flask_server.py

```diff
@@ -171,11 +125,46 @@ def gallery_api(username, date):
         to_render_items = get_folders(user_path)
     else:
         to_render_items = get_folders(folder_path)
-    return jsonify({'items' : to_render_items})
+    return jsonify({'items': to_render_items})
+
+@app.route("/api/settings/", methods=['GET', 'POST'])
+def get_settings_api():
+    if request.method == 'GET':
+        logger.info(f"API/GET request to /settings/")
+        return jsonify(settings.get())
+
+    elif request.method == 'POST':
+        logger.info(f"API/POST request to /settings/")
+        user_settings = settings.get()
+        res = request.get_json()
+        user_settings.update(res)
+        settings.update_settings_file(user_settings)
+        loop_args = {"loop_delay_seconds": int(user_settings["loop_delay_seconds"]),
+                     "loop_variation_percentage": int(user_settings["loop_variation_percentage"])}
+        scraper_runner.updateDelay(**loop_args)
+        return user_settings
 
-@app.route("/logs/", methods=['GET'])
-def logs():
-    return render_template('logs.html', logs=get_system_logs())
+@app.route('/api/loggedin/', methods=["GET"])
```

Comment:
I would not add a new api route for this. Just use api/user/settings and then check client side if the user is logged in.

Even better, enrich the api/user/settings to contains if a user is logged or not if you don't want to place the "session_id" login on the client side.